ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_core.cc

```diff
@@ -443,52 +451,53 @@ TFCore::run (const GstTensorMemory * input, GstTensorMemory * output)
   Tensor in;
 
   for (int i = 0; i < inputTensorMeta.num_tensors; ++i) {
-
-    if (mem_optmz) {
-      TFBuffer *buf = new TFBuffer;
-      buf->len_ = input[i].size;
-      buf->data_ = input[i].data;
-
-      TF_DataType dataType = getTensorTypeToTF_Capi (input[i].type);
-
-      if (dataType == TF_RESOURCE){
-        g_critical ("This data type is not valid: %d", input[i].type);
-        buf->Unref();
-        return -1;
-      }
-
-      in = TensorCApi::MakeTensor (
-        dataType,
-        input_tensor_info[i].shape,
-        buf
-      );
-      if (!in.IsAligned ()) {
-        g_critical ("the input tensor %s is not aligned", inputTensorMeta.info[i].name);
-        buf->Unref();
-        return -2;
-      }
+    /* If the datatype is STRING, it should be handled in specific process */
+    if (input_tensor_info[i].type == DT_STRING) {
+      in = Tensor (input_tensor_info[i].type, input_tensor_info[i].shape);
+      in.scalar<string>()() = string ((char *) input[i].data, input[i].size);
     }
     else {
-      in = Tensor (input_tensor_info[i].type, input_tensor_info[i].shape);
-
-      /* copy data */
-      if (input_tensor_info[i].type == DT_STRING) {
-        in.scalar<string>()() = string ((char *) input[i].data, input[i].size);
-      } else {
+      if (mem_optmz) {
+        TFBuffer *buf = new TFBuffer;
+        buf->len_ = input[i].size;
+        buf->data_ = input[i].data;
+
+        TF_DataType dataType;
+        if (!getTensorTypeToTF_Capi (input[i].type, &dataType)){
+          g_critical ("This data type is not valid: %d", input[i].type);
+          buf->Unref();
+          return -1;
+        }
+        /* this input tensor should be UNREF */
+        in = TensorCApi::MakeTensor (
+          dataType,
+          input_tensor_info[i].shape,
+          buf
+        );
+        if (!in.IsAligned ()) {
+          g_critical ("the input tensor %s is not aligned", inputTensorMeta.info[i].name);
+          buf->Unref();
+          return -2;
+        }
+      }
+      else {
```

Comment:
according to the coding convention, it is ```} else {```.

Reply:
it's fixed :)